src/keep_gpu/single_gpu_controller/cuda_gpu_controller.py

```
@@ -154,21 +182,45 @@ def __exit__(self, exc_type, exc, tb):
+            logger.error("%s", exc)
+            if startup_errors is not None:
+                startup_errors.append(exc)
+            if startup_evt is not None:
+                startup_evt.set()
             return
         assert stop_evt is not None
 
-        torch.cuda.set_device(self.rank)
+        try:
+            torch.cuda.set_device(self.rank)
+        except Exception as exc:
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)